app/serializers/api/v1/item_serializer.rb

```diff
@@ -3,26 +3,16 @@ class ItemSerializer < ApplicationSerializer
     embed :ids, include: true
 
     attributes :id, :donor_description, :state, :offer_id, :reject_reason,
-      :created_at, :updated_at, :package_type_id, :message_ids, :saleable,
+      :created_at, :updated_at, :package_type_id, :saleable,
       :rejection_comments, :donor_condition_id, :rejection_reason_id
 
     has_many :packages, serializer: PackageSerializer
+    has_many :messages, serializer: MessageSerializer, polymorphic: true
```

Comment:
I'm not sure about this change... this removes the ability to filter out the non_private and private messages from Items. Can you ping me about his @bharat619 

Also, we don't want to include messages in the data payload for items, but moving them to a `has_many` means they will be included and this will dramatically slow down offer search (which loads items and therefore now, messages)

We should be able to add back in the `message_ids` reference with the following SQL

```
    def message_ids	
      User.current_user.try(:donor?) ? object.messages.non_private.pluck(:id) : object.messages.pluck(:id)	
    end	

    def message_ids__sql	
      need_private = User.current_user.try(:donor?) ?	
        "AND messages.is_private = 'f'" : ""	
      "coalesce((select array_agg(messages.id) from messages where	
        messageable_type = 'Item' AND messageable_id = items.id #{need_private}), '{}'::int[])"	
    end	
```